R/dm.R

```diff
@@ -493,3 +484,7 @@ cdm_rename_tbl <- function(dm, ...) {
     .init = dm
   )
 }
+
+all_same_source <- function(tables) {
+  all(map_lgl(tables, same_src, tables[[1]]))
```

Comment:
- `detect()` aborts on first match
- Introduce pattern to deal with 0-length list

```suggestion
  # Use `NULL` if `tables` is empty
  first_table <- tables[1][[1]]
  is.null(detect(tables[-1], ~ same_src(., first_table)))
```